File: src/main/java/example/quicksort/Quicksort.java
package example.quicksort;

import example.quicksort.service.QuicksortService;
import otterop.lang.Array;
import otterop.lang.String;
import otterop.lang.Result;

import static otterop.io.Console.println;

public class Quicksort {

    public static void main(java.lang.String[] args) {
        Array<String> argsArray = Array.wrapString(args);
        QuicksortService.quickSort(argsArray, 0, argsArray.size());
        int i = 0;
        while (i< argsArray.size()) {
            println(argsArray.get(i));
            i = i + 1;
        }
        Result.of(String.wrap(""), String.wrap(""));
    }
}

File: dotnet/Example/Quicksort/Quicksort.cs
namespace Example.Quicksort
{
    class Quicksort
    {
        public static void Main(string[] args)
        {
            Otterop.Lang.Array<Otterop.Lang.String> argsArray = Otterop.Lang.Array.WrapString(args);
            Example.Quicksort.Service.QuicksortService.QuickSort(argsArray, 0, argsArray.Size());
            int i = 0;
            while (i < argsArray.Size())
            {
                Otterop.Io.Console.Println(argsArray.Get(i));
                i = i + 1;
            }

        }

    }

}
